test/http-api/index.js

```diff
@@ -19,19 +18,35 @@ describe('HTTP API', () => {
 
   let http = {}
 
-  before(function (done) {
-    this.timeout(60 * 1000)
-
+  const startHttpAPI = (cb) => {
     const options = {
       pass: hat(),
       enablePubsubExperiment: true
     }
     http.api = new API(repoTests, null, options)
 
     ncp(repoExample, repoTests, (err) => {
-      expect(err).to.not.exist()
+      if (err) {
+        return cb(err)
+      }
+
+      http.api.start(false, (err) => {
+        if (err) {
+          return cb(err)
+        }
+        cb(null, http)
+      })
```

Comment:
@VictorBjelkholm this could be using the js-ipfsd-ctl now. No need a different custom setup.

Reply:
Let's open a issue to track this, to try to keep unrelated changes away from this PR.

Reply:
@diasdavid @dryajov you disagree with each other. @diasdavid does it not make sense to keep changes in this branch as small as possible? The purpose is to add jenkins with as small changes as possible. Doing a unrelated refactor in the middle of that feels... Too much.

Reply:
I agree with @VictorBjelkholm I think this is not a critical change for this PR which already seems pretty large and involved. I see no harm in documenting this as a separate issue and handling it elsewhere.

Reply:
I disagree. Look above, there was a bunch of code added that would disappear if this suggestion would be taken.